docker-app/qfieldcloud/core/admin.py

```diff
@@ -982,29 +1016,54 @@ def set_status_unpermitted(self, request, queryset):
 
     def response_change(self, request, delta):
         if "_apply_delta_btn" in request.POST:
-            if not delta.project.project_filename:
-                self.message_user(request, "Missing project file")
-                raise exceptions.NoQGISProjectError()
-
-            if not jobs.apply_deltas(
-                delta.project,
-                request.user,
-                delta.project.project_filename,
-                delta.project.overwrite_conflicts,
-                delta_ids=[str(delta.id)],
-            ):
-                self.message_user(request, "No deltas to apply")
-                raise exceptions.NoDeltasToApplyError()
-
-            self.message_user(request, "Delta application started")
-
-            # we need to sleep 1 second, just to make sure the apply delta started
-            time.sleep(1)
-
-            return HttpResponseRedirect(".")
+            return self.apply_delta(request, delta)
 
         return super().response_change(request, delta)
 
+    def apply_delta(self, request, delta):
+        if not delta.project.project_filename:
+            self.message_user(request, "Missing project file")
+            raise exceptions.NoQGISProjectError()
+
+        if not jobs.apply_deltas(
+            delta.project,
+            request.user,
+            delta.project.project_filename,
+            delta.project.overwrite_conflicts,
+            delta_ids=[str(delta.id)],
+        ):
+            self.message_user(request, "No deltas to apply")
+            raise exceptions.NoDeltasToApplyError()
+
+        self.message_user(request, "Delta application started")
+
+        # we need to sleep 1 second, just to make sure the apply delta started
+        time.sleep(1)
+
+        return HttpResponseRedirect(".")
+
+    @admin.action(description="Download deltas as deltafile")
+    def download_deltafile(self, request, queryset: QuerySet) -> HttpResponse:
+        if queryset.values_list("project_id").distinct().count() != 1:
```

Comment:
Throws for me: I tried it with some existing Delta's though
![image](https://github.com/opengisch/qfieldcloud/assets/21113500/08f85f24-504b-4249-b72a-8b8c1304c3f4)

Reply:
Thanks! Changing the code before pushing is never a smart idea :)